### DISCUSSION TITLE
Speed up the antRegistration pipeline

### DISCUSSION THREAD

**Discussion by LGUMI** [ANTsX/ANTs#1502] [Q&A]
Hello all,

Thanks in advance in giving me advice on this issue. I have two brain image that I tried to register together, the fixed image is the segmented/cropped MRI, and the moving image is postmortem brain 3D reconstruction. I first resized/aligned the two images with freeview (a brain image software that is freesurfer based), and run ANTs script on the Freeview results. The rigid part is relatively quick, however, for the BSplineSyN part, it became extremely slow..... I wonder if that is something I can adjust on the parameter side, or should I use other transformation function other than BSplineSyN. I attached the MRI, postmortem 3D reconstruction (with a resampled one), and my ANT script (google drive link) in the post. Thank you very much!

https://drive.google.com/drive/folders/1CHAeHUkoScMU4JTSZoYcJZl2UrUCjbMY?usp=sharing

Best,
Jonathan

**Comment by ntustison** [comment]
Where did you get the antsRegistration parameters used in your script?

**Reply by LGUMI** [reply]
Hello Nick,

Thanks for the timely reply! I got the script from a colleague of mine who used to work in my lab, on a similar form of data (also postmortem brain vs MRI), it worked relatively fast for his, but for mine it becomes really slow.

**Comment by ntustison** [comment]
Sounds like an issue specific to your machine.  Did you compile ANTs from a recent checkout?

**Reply by LGUMI** [reply]
Interesting, I haven't updated/recompile ANTs for a while as I recalled, would you recommend me updating it? And should I provide some info like package version for you to better diagnose/troubleshoot?

As always, thanks a lot for your help!
Jonathan

**Reply by LGUMI** [reply]
ANTs Version: 2.3.3.dev167-gca97e
Compiled: May 19 2020 21:35:44

This is the version of ANTs I am currently at, and I am using Linux Mint 18.2 Cinnamon 64-bit

**Reply by ntustison** [reply]
Yes.  Generally we recommend to update and recompile before posting any issues.  The toolkit is fairly dynamic and prone to frequent updating. Specifically, with respect to the original issue, you might be dealing with this [recent discovery and subsequent fix](https://github.com/ANTsX/ANTs/issues/1017).

**Reply by LGUMI** [reply]
Thank you very much! I will recompile it to see if the issue resolve.

Best,
Jonathan
